src/drive/drive.ts

```diff
@@ -0,0 +1,247 @@
+import { promises as fs } from 'fs';
+
+import url from '../constants/url';
+import Requests from '../utils/request';
+import { isString } from '../utils/string';
+import { DriveApi } from '../constants/api';
+import { ObjectType } from '../types/basic';
+import { PutOptions, ListOptions } from '../types/drive/request';
+
+import {
+  GetResponse,
+  PutResponse,
+  ListResponse,
+  UploadResponse,
+  DeleteResponse,
+  DeleteManyResponse,
+} from '../types/drive/response';
+
+export default class Drive {
+  private requests: Requests;
+
+  /**
+   * Drive constructor
+   *
+   * @param {string} projectKey
+   * @param {string} driveName
+   */
+  constructor(projectKey: string, driveName: string) {
+    const baseURL = url.DRIVE_HOST_URL.replace(':drive_name', driveName);
+    this.requests = new Requests(projectKey, baseURL);
+  }
+
+  /**
+   * get file from drive
+   *
+   * @param {string} name
+   * @returns {Promise<GetResponse>}
+   */
+  public async get(name: string): Promise<GetResponse> {
+    const trimedName = name.trim();
+    if (!trimedName.length) {
+      throw new Error('Name is empty');
+    }
+
+    const encodedName = encodeURIComponent(trimedName);
+
+    const { status, response, error } = await this.requests.get(
+      DriveApi.GET_FILE.replace(':name', encodedName)
+    );
+    if (status === 404 && error) {
+      return null;
+    }
+
+    if (error) {
+      throw error;
+    }
+
+    return response;
+  }
+
+  /**
+   * delete file from drive
+   *
+   * @param {string} name
+   * @returns {Promise<DeleteResponse>}
+   */
+  public async delete(name: string): Promise<DeleteResponse> {
+    const trimedName = name.trim();
+    if (!trimedName.length) {
+      throw new Error('Name is empty');
+    }
+
+    const payload: ObjectType = {
+      names: [name],
+    };
+
+    const { response, error } = await this.requests.delete(
+      DriveApi.DELETE_FILES,
+      payload
+    );
+    if (error) {
+      throw error;
+    }
+
+    return response?.deleted?.[0] || name;
```

Comment:
curious what the `?` does after a variable name in TypeScript? Never seen this syntax before and didn't find the docs.

Reply:
check this out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining